File: CPP version/OrcChief.cpp
#include "OrcChief.h"


OrcChief::OrcChief() :Entity()
{
	mace = 40;
	stomp = 60;
	MaxHealth = 200;
	rage = 0;
	CurrentHealth = MaxHealth;
	CoinDrop = 1000;
	InstType = OrcChiefType;
}


OrcChief::~OrcChief()
{
	std::cout << "Deleted the chief" << std::endl;
}


int OrcChief::GetType()
{
	return InstType;
}

void OrcChief::PrintType() const
{
	std::cout << "orc chief";
}

void OrcChief::Print() const
{
	std::cout << "This orc chief`s mace deals " << mace << " damage and he has " << MaxHealth << "HP" << std::endl;
}


int OrcChief::Attack()
{
	if (rage % 5 != 0)
	{
		std::cout << "The chief hits for:" << mace;
		rage++;
		return mace;
	}
	else
	{
		std::cout << "The chief gets really angry and does a ground-breaking stomp dealing " << stomp << " damage";
		rage++;
		return stomp;
	}
}


int OrcChief::Defend(int DmgTaken)
{
	CurrentHealth -= DmgTaken;
	if (CurrentHealth <= 0)
	{
		std::cout << std::endl << "The Chief died " << std::endl;
		return 0;
	}
	std::cout << " and the chiefs hp now is " << CurrentHealth << std::endl;
	return CurrentHealth;
}



int OrcChief::GetCoins()
{
	return CoinDrop;
}

File: Java version/OrcChief.java
public class OrcChief extends Entity{

    private int Mace;
    private int Stomp;
    private int Rage;
    private int CoinDrop;


    public OrcChief(int Mace, int Stomp, int Rage, int CoinDrop, int MaxHealth) {
        super(Type.OrcChief, MaxHealth);
        this.Mace = Mace;
        this.Stomp = Stomp;
        this.Rage = Rage;
        this.CoinDrop = CoinDrop;
    }
    

    //text output methods
    public void PrintType() {
        System.out.println("orc chief");
    }

    public void Print() {
        System.out.println("The orc chief's mace deals " + Mace + 
        " damage and he has " + MaxHealth + " HP");
    }

    //getters and setters
    public Type GetType() {
        return InstType;
    }

    public int GetCoins() {
        return CoinDrop;
    }

    public void AddCoins(int Amount) {
        return;
    }

    //gameplay methods
    public int Attack() {
        if (Rage % 5 != 0) {
            System.out.print("The chief hits for " + Mace);
            Rage++;
            return Mace;
        }
        else {
            System.out.print("The chief gets really angry and stomps the ground dealing " +
            Stomp + " damage");
            Rage++;
            return Stomp;
        }
    }


    public int Defend(int Dmgtaken) {
        CurrentHealth -= Dmgtaken;
        if (CurrentHealth <= 0) {
            System.out.println(" and the chief dies!");
            return 0;
        }
        System.out.println(" and the chief's HP is now " + CurrentHealth);
        System.out.println("");
        return CurrentHealth;
    }

    public void UseItem(ItemType Type, int Effect) {
        
    }
    
}
